webapp/src/modules/mapBuilder.js

```diff
@@ -214,28 +215,34 @@ export default class MapBuilder {
         if (!loc) {
           loc = e.latlng;
           this._mcg.clearLayers();
-          this._map.setView([e.latitude + this._latOffset, e.longitude], 20);
+          this._map.setView([e.latitude + this._latOffset, e.longitude], this._maxZoom);
           line = this._l.polyline([[e.latlng.lat, e.latlng.lng]]);
-          this._tripMarker = this._l
-            .animatedMarker(line.getLatLngs(), {
-              icon: this._scooterIcon,
-            })
-            .addTo(this._map);
+          this._tripMarker = this._l.animatedMarker(line.getLatLngs(), {
+            icon: this._scooterIcon,
+            autoStart: false,
+            duration: 250,
+            distance: 0,
+          });
+          this._map.addLayer(this._tripMarker);
           onGetLocation(e);
         }
         if (
           this._tripMarker &&
           loc &&
           line &&
-          loc.lat !== e.latitude &&
-          loc.lng !== e.longitude
+          (loc.lat !== e.latitude || loc.lng !== e.longitude)
         ) {
+          this._tripMarker.stop();
+          // Sets next location distance for animation purpose
+          const nextDistance = this._l
+            .latLng(loc.lat, loc.lng)
+            .distanceTo([e.latitude, e.longitude]);
+          this._tripMarker.options.distance = nextDistance;
           line.addLatLng([e.latitude, e.longitude]);
           this._tripMarker.start();
-          this._map.setView([e.latitude + this._latOffset, e.longitude], 20, {
+          this._map.flyTo([e.latitude + this._latOffset, e.longitude], this._maxZoom, {
```

Comment:
`tripMarker.start()` and `.flyTo` animations are interfering together, we need to set a ticket to fix this. Probably need to start either animation before the other to prevent this from happening, otherwise lets look into other solutions. @rgalanakis